### PR TITLE
GatewayToBackend API: add otap target related changes

### PR DESCRIPTION
**Pull request by GwendalRaoul**
Doc must be updated later to have permanent link (from master sha1)

### CODE REVIEW

File: gateway_to_backend/protocol_buffers_files/config_message.proto [hunk 1]
```diff
@@ -98,9 +98,12 @@ message GatewayInfo {
...
-    // This field must be explicitely set from code and must be
-    // the one defined in each released verion.
-    // Current version is 1
+    // This field must be explicitly set from code and must be
+    // the one defined in each released version.
+    // Current version is 2
+    // Version 1: Initial release
+    // Version 2: Addition of setScratchpadTargetAndAction
+    // It should be replaced by a feature list
     optional uint32 implemented_api_version = 4;
```
Comment (vgeneves) [inline gateway_to_backend/protocol_buffers_files/config_message.proto:107]: I don't understand comment says version 2. But code is 4 ?

File: gateway_to_backend/protocol_buffers_files/generic_message.proto [hunk 2]
```diff
@@ -23,6 +23,8 @@ message WirepasMessage {
     optional ProcessScratchpadResp process_scratchpad_resp = 14;
     optional GetGwInfoReq get_gateway_info_req = 15;
     optional GetGwInfoResp get_gateway_info_resp = 16;
+    optional SetScratchpadTargetAndActionReq set_scratchpad_target_and_action_req = 17;
+    optional SetScratchpadTargetAndActionResp set_scratchpad_target_and_action_resp = 18;
```
Comment (vgeneves) [inline gateway_to_backend/protocol_buffers_files/generic_message.proto:27]: You don't add a getScratchpad .. ?
(Is there a get Appconfig ?, or is this included in the get gw/sink config ?)

File: gateway_to_backend/protocol_buffers_files/otap_message.proto [hunk 3]
```diff
@@ -18,12 +18,49 @@ enum ScratchpadStatus {
...
 }
 
+enum ScratchpadAction {
+    // Will be the default value in case it is later extended
+    UNKNOWN_ACTION = 0;
+    // No propagation, no processing of scratchpad
+    NO_OTAP = 1;
+    // Propagate the target scratchpad but no processing
+    PROPAGATE_ONLY = 2;
+    // Propagate the target scratchpad and process it when available
```
Comment (vgeneves) [inline gateway_to_backend/protocol_buffers_files/otap_message.proto:28]: when available is a bit confusing. PROCESS immediately or as soon it is received ?

File: gateway_to_backend/protocol_buffers_files/otap_message.proto [hunk 4]
```diff
@@ -18,12 +18,49 @@ enum ScratchpadStatus {
...
+
 message ScratchpadInfo {
     required uint32 len = 1;
     required uint32 crc = 2;
     required uint32 seq = 3;
 }
 
+message TargetScratchpadAndAction {
+    required ScratchpadAction action = 1; // What is the action to perform with the current scratchpad
+    optional uint32 target_sequence = 2; // Between 0 and 255 (if missing using local scratchpad sequence if present)
```
Comment (vgeneves) [inline gateway_to_backend/protocol_buffers_files/otap_message.proto:56]: Between 1 and 254.
I think uploading a scratchpad 255 in a Sink won't work.
And if I am correct, the single mcu api will return an error

File: gateway_to_backend/protocol_buffers_files/config_message.proto (hunk 1 above)
Comment (GwendalRaoul) [reply gateway_to_backend/protocol_buffers_files/config_message.proto:107]: 4 is protobuf field id not value

File: gateway_to_backend/protocol_buffers_files/generic_message.proto (hunk 2 above)
Comment (GwendalRaoul) [reply gateway_to_backend/protocol_buffers_files/generic_message.proto:27]: Already part of getStatus

**Review by vgeneves — APPROVED** [review]

**Pull request merged**